src/main_thread/init/utils/update_manifest_codec_support.ts

```diff
@@ -0,0 +1,190 @@
+    isSupportedEncrypted: boolean | undefined;
+  } => {
+    const inputCodec = `${mimeType ?? ""};codecs="${codec ?? ""}"`;
+    const baseData = codecSupportMap.get(inputCodec);
+    if (baseData !== undefined) {
+      return baseData;
+    }
+
+    let newData;
+    const isSupported = isCodecSupported(inputCodec);
+    if (!isSupported) {
+      newData = {
+        isSupportedClear: false,
+        isSupportedEncrypted: false,
+      };
+    } else if (isNullOrUndefined(contentDecryptor)) {
+      newData = {
+        isSupportedClear: true,
+        // This is ambiguous. Less assume that with no ContentDecryptor, an
+        // encrypted codec is supported
+        isSupportedEncrypted: true,
+      };
+    } else if (contentDecryptor.getState() === ContentDecryptorState.Initializing) {
+      newData = {
+        isSupportedClear: true,
+        isSupportedEncrypted: undefined,
+      };
+    } else {
+      newData = {
+        isSupportedClear: true,
+        isSupportedEncrypted:
+          contentDecryptor.isCodecSupported(mimeType ?? "", codec ?? "") ?? true,
+      };
+    }
+    codecSupportMap.set(inputCodec, newData);
+    updatedCodecs.push({
+      codec: codec ?? "",
+      mimeType: mimeType ?? "",
+      supported: newData.isSupportedClear,
+      supportedIfEncrypted: newData.isSupportedEncrypted,
+    });
+    return newData;
+  };
+
+  manifest.periods.forEach((p) => {
+    [
+      ...(p.adaptations.audio ?? []),
+      ...(p.adaptations.video ?? []),
+      ...(p.adaptations.text ?? []),
+    ].forEach((adaptation) => {
+      let hasSupportedCodec: boolean = false;
+      let hasCodecWithUndefinedSupport: boolean = false;
+      adaptation.representations.forEach((representation) => {
+        if (representation.isSupported !== undefined) {
+          if (representation.isSupported) {
+            hasSupportedCodec = true;
+          }
+          // We already knew the support for that one, continue to next one
+          return;
+        }
+
+        const isEncrypted = representation.contentProtections !== undefined;
+        const mimeType = representation.mimeType ?? "";
+        let codecs = representation.codecs ?? [];
+        if (codecs.length === 0) {
+          codecs = [""];
+        }
+        for (const codec of codecs) {
+          const codecSupportInfo = efficientlyGetCodecSupport(mimeType, codec);
+          if (!isEncrypted) {
+            representation.isSupported = codecSupportInfo.isSupportedClear;
+          } else if (
+            representation.isSupported !== codecSupportInfo.isSupportedEncrypted
+          ) {
+            representation.isSupported = codecSupportInfo.isSupportedEncrypted;
+          }
+
+          if (representation.isSupported === undefined) {
+            hasCodecWithUndefinedSupport = true;
+          } else if (representation.isSupported) {
+            hasSupportedCodec = true;
+            representation.codecs = [codec];
+          }
+        }
```

Comment:
I think there may be a rare case when there is an array of 2 codecs and one is supported on the other not:
`reprensentation.codecs = [  "hvc1", “dvh1"]`
Let's say dvh1 is not supported and hvc1 is supported.
From my understanding at the end of this loop the representation has `codecs: [hvc1]` and `supported: false`
The solution is probably to add a `break` statement in the last condition.
This case should not really appears because we put the more restrictive codecs first, but if things change in the future it could be a bit difficult to catch it.